### ISSUE TITLE
Swatches not showing on homepage New products widget

### ISSUE THREAD

**Issue by gagan67commerce** [magento/magento2#14134]
Create New products via magento's default widget and call on homepage

### Preconditions
1. Magento 2.4-develop

### Steps to reproduce

1. Create a widget and call New products on hompage 
2. Add to list any configurable product

### Expected result
1. Swatches should be visible on homepage 

### Actual result
1. No swatches coming on homepage
![#14134issue](https://user-images.githubusercontent.com/51681379/71966134-b28e8300-3209-11ea-8714-81069aa632ce.jpg)

**Comment by paragpadsumbiya** [comment]
@gagan67commerce Thanks for reporting .Can you please check by login with sample customer because it might be posible that some widget can be viewed if customer has logged in .You need to change that permission .

**Comment by gagan67commerce** [comment]
No, the issue is not with the logged in, the issue actually is the file which is rendering the block does not have the block for swatches rendered. What basically is needed is that the render.block should be there within the catalogwidget module, which is how the product swatches are there in the list page. It is the code that is missing.

**Comment by magento-engcom-team** [comment]
@gagan67commerce, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by athulvijayan** [comment]
"I am working on this at #dmcdindia”

**Comment by magento-engcom-team** [comment]
@athulvijayan thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by ashokramji** [comment]
I am working on this at #dmcdindia

**Comment by magento-engcom-team** [comment]
@ashokramji thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by jignesh-baldha** [comment]
I am working on this at #14663

**Comment by jignesh-baldha** [comment]
I am working on this at #dmcdindia

**Comment by magento-engcom-team** [comment]
@jignesh-baldha thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by kerlama** [comment]
Hi,

any progress on this one, as this is not working on 2.2.5 too.

In fact, if you add configurable product to widget and then click on add to cart button, it will go to empty cart page.

**Comment by magento-engcom-team** [comment]
Hi @rav-redchamps. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if you want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @eduard13. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if you want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by sunnykhatri** [comment]
I am working on this at #dmcdindia19

**Comment by magento-engcom-team** [comment]
@sunnykhatri thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by benboudan** [comment]
Hello 

Have you any news about this request ?

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-30276` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by lavanyadckap** [comment]
#mm20in

**Comment by magento-engcom-team** [comment]
@lavanyadckap thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by engcom-Bravo** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Bravo** [comment]
Hi @gagan67commerce,

Thanks for your reporting and collaboration.

Verified the issue in Latest 2.4-develop instance and the issue is still reproducible.Kindly refer the screenshots.


**Steps to reproduce**

- Create a widget and call New products on hompage
- Add to list any configurable product



<img width="637" alt="Screenshot 2024-05-02 at 11 52 25" src="https://github.com/magento/magento2/assets/51680745/3229ec2d-2840-483a-bc08-d935f5573ba7">


Swatches are not showing on products listing page and it is working fine for products details page.

Hence Confirming the issue.

Thanks.

**Comment by magentoabu** [comment]
@engcom-Bravo  Can't able to reproduce the issue .please check and close this issue

![Image](https://github.com/user-attachments/assets/b6b06a96-de82-424b-be35-9e89c1d33591)

**Comment by engcom-Bravo** [comment]
@magentoabu Thanks for your Contribution!!.

Hi @gagan67commerce,

Thanks for your reporting and collaboration.

we have tried to reproduce the issue in Latest 2.4-develop instance and the issue is no more reproducible.Kindly refer the screenshots.


<img width="1436" alt="Image" src="https://github.com/user-attachments/assets/0f7fd55f-4216-44fb-9961-484a46922fda" />


Hence we are closing this issue.


Thanks.
